### ISSUE TITLE
espnet2 aishell recipe

### ISSUE THREAD

**Issue by 200987299** [espnet/espnet#3290]
Dear developers, I am tring to write a script for aishell2 from aishell1 script, but when it comes to training process, I always get NaN error. (grad is NaN, skiping update model). It happened for all transformer recipes in the conf/tuning/ folder. Does anyone have any  idea how to solve this problem?

thanks!

**Comment by sw005320** [comment]
How often does it happen?
If it does not happen so often, we can ignore it.

**Comment by 200987299** [comment]
> How often does it happen?
> If it does not happen so often, we can ignore it.

Dear Professor Watanabe,

It happens every time for all transformer recipes. I am tuning the learning rate hoping to get a better result. I am also trying the rnn recipes, let you know later.

Bests
D

**Comment by sw005320** [comment]
It should not happen so often. I’m curious about your config files.
Maybe, it is a good time to make an aishell2 recipe in espnet2.
Is it OK for you to make a PR about your current one, then do some debugging together through PR? 
Or I may ask some of my colleagues to do it.
(@pengchengguo and @yuekaizhang, I remember that either of you touched aishell2. Do you think you can make an espnet2 recipe?)

**Comment by 200987299** [comment]
> It should not happen so often. I’m curious about your config files.
> Maybe, it is a good time to make an aishell2 recipe in espnet2.
> Is it OK for you to make a PR about your current one, then do some debugging together through PR?
> Or I may ask some of my colleagues to do it.
> (@pengchengguo and @yuekaizhang, I remember that either of you touched aishell2. Do you think you can make an espnet2 recipe?)

I would love to participate. I copied 99% of the code from espnet2/aishell. I am starting to think that the error happened because I didn't filter out un-clean data from the dataset. I will post my code a bit later after I check it again. I am also love to work with @pengchengguo and @yuekaizhang, thanks for the help and input.

**Comment by sw005320** [comment]
Very cool.
Actually, debugging through PR or issues would work.
So, if you’re ready, please make a PR.
We may find an issue quickly.

**Comment by kamo-naoyuki** [comment]
Recently we found a fatal bug, #3283, and this bug was existing for a long time.  This is my fault, sorry.  If you didn't update espnet, please do it, and I recommend you using `--ctc_conf ignore_nan_grad= true`

**Comment by 200987299** [comment]
> Recently we found a fatal bug, #3283, and this bug was existing for a long time. This is my fault, sorry. If you didn't update espnet, please do it, and I recommend you using `--ctc_conf ignore_nan_grad= true`

Thanks, going to try it now.

**Comment by pengchengguo** [comment]
I also get the `nan grad` problem when running Switchboard corpora and it seems because the input length is shorter than the target label. I add an additional CTC parameter in the config file, like
```
ctc_conf:
    ignore_nan_grad: true
```
 and all add a code the monitor how many errors are caused by the length problem. Maybe you can try it too.

**Comment by sw005320** [comment]
Maybe, we can make `ignore_nan_grad: true` as default?
Actually, I'm always using this.

**Comment by pengchengguo** [comment]
Yes, it's better to set the `ignore_nan_grad: true` as default. I can do it or maybe @200987299 can fix it in his aishell2 recipe PR.

**Comment by 200987299** [comment]
> Yes, it's better to set the `ignore_nan_grad: true` as default. I can do it or maybe @200987299 can fix it in his aishell2 recipe PR.

I am testing it now, will do it after I finished train 2 epochs. (just to make sure it works)
